worlds/sc2/rules.py

```diff
@@ -2249,6 +2372,18 @@ def enemy_intelligence_third_stage_requirement(self, state: CollectionState) ->
                 )
             )
         )
+    
+    def enemy_intelligence_cliff_garrison_and_nova_mobility(self, state: CollectionState) -> bool:
+        return (
+            self.enemy_intelligence_cliff_garrison(state)
+            and (
+                self.nova_any_nobuild_damage(state)
+                or (
+                    state.has(item_names.NOVA_PROGRESSIVE_STEALTH_SUIT_MODULE, self.player, 2)
```

Comment:
This isn't processed well by `ValidInventory`

Reply:
Updated `ValidInventory` and `TestInventory` to handle this. I figured it would be nicer if future logic writers could use this without having to know about the gotcha.